pkg/operatorsource/configuring.go

```diff
@@ -115,38 +113,24 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	packages := r.datastore.GetPackageIDsByOperatorSource(out.GetUID())
 	out.Status.Packages = packages
 
-	cscCreate := new(builders.CatalogSourceConfigBuilder).WithTypeMeta().
-		WithNamespacedName(in.Namespace, in.Name).
-		WithLabels(in.GetLabels()).
-		WithSpec(in.Namespace, packages, in.Spec.DisplayName, in.Spec.Publisher).
-		WithOwnerLabel(in).
-		CatalogSourceConfig()
+	grpcCatalogEnsurer := grpccatalog.NewEnsurer(r.logger, r.reader, r.client)
 
-	err = r.client.Create(ctx, cscCreate)
-	if err != nil && !k8s_errors.IsAlreadyExists(err) {
-		r.logger.Errorf("Unexpected error while creating CatalogSourceConfig: %s", err.Error())
-		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
-		return
+	key := client.ObjectKey{
+		Name:      in.Name,
+		Namespace: in.Namespace,
 	}
-
-	if err == nil {
-		nextPhase = phase.GetNext(phase.Succeeded)
-		r.logger.Info("CatalogSourceConfig object has been created successfully")
-
-		return
+	// Get all labels from OperatorSource and add the DatastoreLabel to the map
+	labels := map[string]string{
```

Comment:
This should be done inside the builder.

Reply:
Previously we were passing the entire catalogsourceconfig around in order to propagate labels. Now that we no longer create the in-memory catalogsourceconfig as part of the operatorsource configuring reconciler, we need to pass along these labels.